networkx/algorithms/approximation/tests/test_traveling_salesman.py

```diff
@@ -352,19 +359,27 @@ def test_TSP_weighted():
 
     # Check all methods
     methods = [
-        nx_app.christofides,
-        nx_app.greedy_tsp,
-        lambda G, wt: nx_app.simulated_annealing_tsp(G, "greedy", weight=wt),
-        lambda G, wt: nx_app.threshold_accepting_tsp(G, "greedy", weight=wt),
+        (nx_app.christofides, {}),
+        (nx_app.greedy_tsp, {}),
+        (
+            nx_app.simulated_annealing_tsp,
+            {"init_cycle": "greedy"},
+        ),
+        (
+            nx_app.threshold_accepting_tsp,
+            {"init_cycle": "greedy"},
+        ),
     ]
     for method in methods:
-        cycle = tsp(G, nodes=[3, 6], weight="weight", method=method)
+        cycle = tsp(G, nodes=[3, 6], weight="weight", method=method[0], **method[1])
```

Comment:
Maybe slightly easier to read by splitting the `method` into its two parts:
```python
for meth, kwds in methods:
        cycle = tsp(G, nodes=[3, 6], weight="weight", method=meth, **kwds)

<similar for path and tourpath>
```

Reply:
I like this better too!